lib/app.go

```diff
@@ -136,9 +137,26 @@ func appStateInMutablePod(app *v1.App, pod *pkgPod.Pod) error {
 			}
 		}
 	}()
+	createdFile := common.AppCreatedPath(pod.Path(), app.Name)
+	startedFile := common.AppStartedPath(pod.Path(), app.Name)
+	appStatusFile := common.AppStatusPath(pod.Path(), app.Name)
+
+	if pod.UsesOverlay() {
+		treeStoreID, err := pod.GetStage1TreeStoreID()
+		if err != nil {
+			return err
+		}
+		s1Dir := filepath.Join(filepath.Join(pod.Path(), "overlay"), treeStoreID)
+		upper := filepath.Join(s1Dir, "upper")
+		appStatusesPath := filepath.Join(upper, "/rkt/status")
+
+		createdFile = filepath.Join(appStatusesPath, fmt.Sprintf("%s-created", app.Name))
+		startedFile = filepath.Join(appStatusesPath, fmt.Sprintf("%s-started", app.Name))
```

Comment:
I would prefer if it was:
```
	createdFile := common.AppCreatedPathOverlay(pod, app.Name)
	startedFile := common.AppStartedPathOverlay(pod, app.Name)
	appStatusFile := common.AppStatusPathOverlay(pod, app.Name)
```
And add those 3 new functions with a "Overlay" suffix in common/common.go.

Because at the moment, the way to build the flag filename (e.g. "%s-created") is spread over different Golang packages.

Reply:
I implemented a slightly different thing since we already had some logic for dealing with overlay in pods.go.